### PR TITLE
Clean up keys commands

### PR DESCRIPTION
**Pull request by keeramis**
## Description

Remove usage of dfu device specs and reduce the complexity of the keys commands by supporting only one protocol per device. Also remove the need to specify the device id for `particle keys` commands if only one device is connected.

## How to Test

1. Run any `particle keys xxx` commands and `particle doctor` command on your device(s) and verify that they are working as expected. 
1. For example, To run `particle keys xxx`, run `npm start -- keys xxx` from project's root directory

## Related Issues / Discussions

https://app.shortcut.com/particle/story/122385/remove-usage-of-dfu-util-in-particle-cli

## Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed [CLA](https://docs.google.com/a/particle.io/forms/d/1_2P-vRKGUFg5bmpcKLHO_qNZWGi5HKYnfrrkd-sbZoA/viewform)
- [x] Problem and solution clearly stated
- [x] Tests have been provided
- [x] Docs have been updated
- [x] CI is passing

### CODE REVIEW

File: src/cmd/keys.js [hunk 1]
```diff
@@ -617,27 +588,54 @@ module.exports = class KeysCommand {
...
 		} catch (err) {
 			throw new VError(ensureError(err), 'Unable to get DFU device');
 		}
 	}
 
-	_validateSegmentSpecs(segmentName) {
-		let specs = deviceSpecs[this.dfuId] || {};
-		let params = specs[segmentName];
+	_validateSegments(segmentName) {
+		let segments = this._getAllSegments() || {};
```
Comment (hugomontero) [inline src/cmd/keys.js:599]: Given that you don't change `segments` and `params` after creation you can put `const` instead of `let`

Comment (keeramis) [reply src/cmd/keys.js:599]: yes! surprised that lint didn't warn this time

Comment (hugomontero) [reply src/cmd/keys.js:599]: Regarding this https://github.com/particle-iot/particle-cli/pull/675/files#r1340443296

You can do:
```suggestion
		const segments = this._getAllSegments();
		const params = segments ? segments[segmentName] : null;
```

Comment (hugomontero) [reply src/cmd/keys.js:599]: Checking more deep the way getAllSegments works you always will receive an object. there is no way to get a null. So the `|| {}` can be removed

File: src/cmd/flash.js [hunk 2]
```diff
@@ -55,29 +55,46 @@ module.exports = class FlashCommand extends CLICommandBase {
...
-		const resetAfterFlash = !factory && modulesToFlash[0].prefixInfo.moduleFunction === ModuleInfo.FunctionType.USER_PART;
-		await flashFiles({ device, flashSteps, resetAfterFlash, ui: this.ui });
+			this.ui.write(`Flashing ${platformName} device ${device.id}`);
+			const resetAfterFlash = !factory && modulesToFlash[0].prefixInfo.moduleFunction === ModuleInfo.FunctionType.USER_PART;
+			await flashFiles({ device, flashSteps, resetAfterFlash, ui: this.ui });
+		} catch (error) {
+			if (device) {
+				await device.close();
+			}
+		}
```
Comment (keeramis) [inline src/cmd/flash.js:97]: @hugomontero and I figured that device doesn't close and it leaves the terminal hanging if the logic fails in any of the above lines such as `const modulesToFlash = await parseModulesToFlash({ files });` . We are if we should add this try/catch block here in this file for other methods (which is `flashLocal`) or if there is a better way to handle it.

**Review by monkbroc — APPROVED** [review]

**Pull request merged**